src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -86,12 +88,14 @@ internal class LocalGrainDirectory :
         internal readonly CounterStatistic UnregistrationsManyRemoteSent;
         internal readonly CounterStatistic UnregistrationsManyRemoteReceived;
 
-        
         public LocalGrainDirectory(Silo silo)
         {
             log = LogManager.GetLogger("Orleans.GrainDirectory.LocalGrainDirectory");
 
             MyAddress = silo.LocalMessageCenter.MyAddress;
+            isSiloNextInTheRing = (siloAddr, hash, excludeMySelf) => (siloAddr.GetConsistentHashCode() <= hash) &&
+                                                                     (!excludeMySelf || !siloAddr.Equals(MyAddress));
```

Comment:
There's 2 things here. Since this is inlining FindLast, then there is not even a need to remove the delegate allocation, as there is no longer a closure being passed to another higher order function.
But even if we decided to continue using FindLast and remove the inlining, what I don't understand is the need for a function field that captures `this`, since that is exactly what an instance method is. So just define the function as a method body instead of in a field that is later assigned.